### DISCUSSION TITLE
Does this firmware applies any changes on the existing operation settings of the heatpump

### DISCUSSION THREAD

**Discussion by m-baumgartner** [gekkekoe/esphome-ecodan-hp#89] [Announcements]
Hello,
I'm in the process of preparing the ESP and i'm wondering what exactly is going to happen when i plug it in CN105 for the first time.

So far i'm quite happy with the way my heatpump is running and i'm really interested in monitoring further into the details of how it operates and where I can extract more efficiency. I'd like however to not change anything from the current settings for now.

If i flash my esp with this firmware, and then plug it in, will it overwrite the configs with some defaults or will it read the existing configs first and keep them until i manually (from webconsole or hass sensor) select another value?

eg : i'm currently running both my zones (ground floor has UFH, upstairs has radiators) in heat compensation curve, but i can see that the defaults of the firmware are on 'heat target temperature'. When i plug the ESP, will it keep my heat compensation curve mode as well as both my curves?

**Comment by gekkekoe** [comment]
what makes you think heat target temp is the default ?!
The esp tries to read the state and will update the esp, this can take up to 2 mins. But since this is a new install, no previous state is known. I think it will work out, but in the event that something changed, you can easily change it back via the esp.

**Reply by m-baumgartner** [reply]
> what makes you think heat target temp is the default ?!

it's the default in the web ui. (probably just because it's the first item of the list).

i'm probably being over cautious to be honest. thank you for your answer.

**Reply by gekkekoe** [reply]
It's just a starting value, we are reading back state from the heatpump and update UI when we have.

**Comment by DamijanD** [comment]
Even if something changes it will be just some simple parameter (not the the curve itself but maybe only the mode). And these settings can be very quickly changed back. In my case room setpoint temperature got changed to 16° and maybe the dhw setpoint temp. Modes and everything else stayed the same (zone1 uses the adaptive mode and zone2 heat compensation curve).

**Reply by m-baumgartner** [reply]
thank you.

**Comment by m-baumgartner** [comment]
i'll plug it in and try to observe changes in menus before after.
I'll log what i see here and then close the discussion so that my findings are searchable.

**Comment by m-baumgartner** [comment]
to anyone finding this, i tried it and watched settings before / after. none of the settings were altered.
all is good.
